model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -295,38 +295,42 @@ class ReverseChannelsPropagationUp(BackReplacementPattern):
         'Subtract': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Pow': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Convert': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
-
-        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through(node, rc),
+        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_pad(node, rc),
     }
 
     @staticmethod
-    def lift_up_through(node: Node, reverse_channels: Node):
+    def lift_up_through_pad(node: Node, reverse_channels: Node):
         r"""
         BEFORE                       AFTER
 
                                      previous_op
                                           \
         previous_op  previous_op       ReverseChannels  previous_op
                  \     /                           \     /
-                   Node                             Node
+                   Pad                              Pad
                     |                                |
               ReverseChannels                      next_op
                     |
                  next_op
 
-        returns boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        returns two objects:
+        first - boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        second - list of ReverseChannels operations that were produced while propagating reverse_channels up
         """
         if node.is_in_port_connected(0):
             node_input_port_0 = node.in_port(0)
-            reverse_channels_out_npde = reverse_channels.out_port(0).get_connection().get_destination().node
+            reverse_channels_out_nodes = [destination.node for destination in
+                                          reverse_channels.out_port(0).get_connection().get_destinations()]
             reverse_channels.out_port(0).disconnect()
-
+            reverse_channels.in_port(0).disconnect()
             src = node_input_port_0.get_connection().get_source()
             node_input_port_0.get_connection().set_source(reverse_channels.out_port(0))
             src.connect(reverse_channels.in_port(0))
-            node.out_port(0).get_connection().set_destination(reverse_channels_out_npde.in_port(0))
-            return True
-        return False
+            for reverse_channels_out_node in reverse_channels_out_nodes:
+                node.out_port(0).connect(reverse_channels_out_node.in_port(0))
```

Comment:
this is absolutely incorrect written. Destination port index is not necessarily equal to zero. It can be any.
So I propose to:.
```
for reverse_channels_destination in reverse_channels.out_port(0).get_connection().get_destinations():
    node.out_port(0).get_connection().add_destination(reverse_channels_destination)
```

Reply:
The propose is not correct. As result we got nodes without connected destination.

Reply:
![image](https://user-images.githubusercontent.com/29801542/132341038-ae8e9b91-d19e-40a4-9aef-e212fa78991d.png)